### PR TITLE
[4391] Edit Site dialog does not reset state after closing it

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4391

### CODE REVIEW

File: ui/app/src/services/sites.ts [hunk 1]
```diff
@@ -55,7 +56,12 @@ export function deleteSite(id: string) {
 }
 
 export function updateSite(site: Site) {
-  return postJSON(`/studio/api/2/sites/${site.id}`, { name: site.name, description: site.description });
+  return postJSON(`/studio/api/2/sites/${site.id}`, { name: site.name, description: site.description })
+    .pipe(
+      catchError((error: AjaxError) => {
+        throw error.response?.response ?? error;
+      })
```
Comment (rart) [inline ui/app/src/services/sites.ts]: what's this for?

File: ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx [hunk 2]
```diff
@@ -144,13 +160,12 @@ function EditSiteDialog(props: EditSiteDialogProps) {
...
-      aria-labelledby="editSiteDialogTitle"
-      fullWidth
-      maxWidth="sm"
-    >
+    <>
+      <DialogHeader
+        id="editSiteDialogTitle"
+        title={<FormattedMessage id="editSiteDialog.title" defaultMessage="Edit Site" />}
+        onDismiss={onClose}
+      />
```
Comment (rart) [inline ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx]: What's the purpose for this change?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx (hunk 2 above)
Comment (jvega190) [reply ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx]: Moved to the original component and handled error using ErrorDialog

File: ui/app/src/services/sites.ts (hunk 1 above)
Comment (jvega190) [reply ui/app/src/services/sites.ts]: Reverted, error is now being passed to the ErrorDialog without this

**Review by rart — CHANGES_REQUESTED** [review]
Please fix conflicts

File: ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx [hunk 3]
```diff
@@ -219,6 +227,7 @@ function EditSiteDialogUI(props: EditSiteDialogUIProps) {
       <DialogHeader
         id="editSiteDialogTitle"
         title={<FormattedMessage id="editSiteDialog.title" defaultMessage="Edit Site" />}
+        onDismiss={onClose}
```
Comment (rart) [inline ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx]: should be using onDismiss prop

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Comment by rart** [comment]
Changes applied on https://github.com/craftercms/studio-ui/pull/2020

@jvega190 please fix event props typings once merged.
